plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -703,6 +703,16 @@ private Answer execute(ResizeVolumeCommand cmd) {
             }
 
             VirtualDisk disk = vdisk.first();
+            if(vdisk.second()!=null && !vdisk.second().toLowerCase().contains("scsi"))
```

Comment:
@priyankparihar _vdisk.second()_ here holds the disk bus name, that would ide0:0,... (or) scsi0:0,...
I think startsWith("scsi") check would be more precise.

Move "_VirtualDisk disk = vdisk.first();_" stmt below this if check. For all non-scsi disks, the _disk_ var is unnecessarily stacked.